frontend/src/components/Registration.js

```diff
@@ -0,0 +1,275 @@
+import { getCookie } from "../utils.js";
+import { createSignal } from "solid-js";
+import { genderChoices, stateChoices } from "../constants.js";
+import { useNavigate } from "@solidjs/router";
+
+const RegistrationForm = () => {
+  const csrftoken = getCookie("csrftoken");
+  const [firstname, setFirstName] = createSignal("");
+  const [lastname, setLastName] = createSignal("");
+  const [dateofbirth, setDateOfBirth] = createSignal("");
+  const [phone, setPhone] = createSignal("");
+  const [upaiprofile, setUpaiProfile] = createSignal("");
+  const [teamname, setTeamName] = createSignal("");
+  const [occupation, setOccupation] = createSignal("");
+  const [educationinstitution, setEducationInstitution] = createSignal("");
+  const [gender, setGender] = createSignal("");
+  const [othergender, setOtherGender] = createSignal("");
+  const [notinindia, setNotInIndia] = createSignal(false);
+  const [city, setCity] = createSignal("");
+  const [state, setState] = createSignal("");
+  const navigate = useNavigate();
+
+  // Gender
+  const handleGenderChange = (e) => {
+    setGender(e.target.value);
+    if (e.target.value !== "O") {
+      setOtherGender("");
+    }
+  };
+
+  // Not in India checkbox
+  const handleNotInIndiaChange = () => {
+    setNotInIndia(!notinindia());
+    setState("");
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    // Prepare the data to be submitted
+    const formData = {
+      first_name: firstname(),
+      last_name: lastname(),
+      date_of_birth: dateofbirth(),
+      gender: gender(),
+      other_gender: othergender(),
+      city: city(),
+      // phone: phone(), // FIXME: Uncomment when multiple ModelSchema is implemented
+      state_ut: state(),
+      not_in_india: notinindia(),
+      team_name: teamname(),
+      occupation: occupation(),
+      educational_institution: educationinstitution(),
+      india_ultimate_profile: upaiprofile(),
+    };
+    // Perform form submission logic or API request with formData
+    console.log("Form submitted with data:", formData);
+
+    submitFormData(formData);
+  };
+
+  const submitFormData = async (formData) => {
+    formData.not_in_india = formData.not_in_india.toString();
+    // Send a post request to the api
+    try {
+      const response = await fetch("/api/registration", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "X-CSRFToken": csrftoken },
+        body: JSON.stringify(formData),
+      });
+
+      if (response.ok) {
+        // Handle success
+        console.log("Player created successfully");
+        // Reset form fields
```

Comment:
Why do you need to reset the form fields, if you are navigating away from the registration page?